pubsub.go

```diff
@@ -1164,6 +1175,15 @@ type TopicOptions struct{}
 
 type TopicOpt func(t *Topic) error
 
+// WithCustomMessageIDFunc is an option to customize the way a message ID is computed for a pubsub message
+// within the Topic. By default, it uses global MsgIdFunction registered on a PubSub instance.
+func WithCustomMessageIDFunc(msgId MsgIdFunction) TopicOpt {
```

Comment:
Not sure about a good naming here. Added a work `Custom` to differ from global Pubsub option.